### PR TITLE
Add SVG Sprite Module for icons

### LINKED ISSUE #985: Use sprites for SVG icons
## Problem

Using one file per an SVG icon requires the browser to make many requests to get all the icons. This is wasteful, and sometimes results in icons not being loaded. You can see this when CI VR tests fail: 
content-switcher-with-external-sources-open-ltr-xs-diff

## Description

We should use a sprite for SVG icons. This will mean that the user loads a slightly larger file once, and then it is reused throughout the site.
We should keep the single SVGs in a directory, and have a script that creates the sprite and updates the it if the SVGs are added/removed/changed.

## Additional context

We should implement it in a TS-friendly way so that svgs in components don't create TS errors anymore.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #985 by @obulat

## Description

This PR adds the Nuxt [SVG Sprite Module](https://github.com/nuxt-community/svg-sprite-module) for icons. 

To minimize changes, I wrapped the `SvgIcon` component provided by the module in the `VIcon` component.

This PR also adds documentation page about the icons. I'm not sure where it should be placed.

## Testing Instructions

All the snapshot tests should pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/test/unit/setup.js [hunk 1]
```diff
@@ -21,6 +22,12 @@ config.stubs["nuxt-link"] = Vue.component("NuxtLink", {
   template: '<a :href="to" v-on="$listeners"><slot /></a>',
 })
 
+config.stubs["svg-icon"] = Vue.component("SvgIcon", {
+  props: ["name"],
+  template: '<svg title="name" />',
+})
```
Comment (sarayourfriend) [inline frontend/test/unit/setup.js:28]: Are we able to remove the [Jest SVG transformer](https://github.com/WordPress/openverse/blob/697f62f01a32cb7fcf2f4a7627650a113cba40da/frontend/jest.config.js#L22) after this?

File: documentation/guides/frontend/icons.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+The name prop is used to identify the icon within the SVG sprite. You can find
+the names of the available icons in the `frontend/src/assets/icons/sprites.json`
+file. The default icons listed in the object with the name `icons` use the name
+as is, without prefixes. To use the license icons, you would need to add
+`license/` as a prefix to the name. For example, to use an `external-link` icon,
+you would use the name `external-link`, and for the `cc-by` icon, you would use
+the name `license/cc-by`.
+
+You can also view a list of all available icons if you go to
+`http://localhost:8443/_icons`. You can copy the icon name (with the prefix, if
```
[the remark this answers is not in the document]
Comment (sarayourfriend) [reply documentation/guides/frontend/icons.md:40]: Okay, I found the fix. Took digging around the nuxt storybook and svg-sprite module code to figure it out. It's a simple path resolution issue but has an obscure fix :cry: 

```diff
diff --git a/frontend/.storybook/main.js b/frontend/.storybook/main.js
index 94e745799..51a0e519c 100644
--- a/frontend/.storybook/main.js
+++ b/frontend/.storybook/main.js
@@ -1,6 +1,6 @@
 const { nuxifyStorybook } = require("../.nuxt-storybook/storybook/main")
 
-module.exports = nuxifyStorybook({
+const storybook = nuxifyStorybook({
   webpackFinal(config) {
     // extend config here
 
@@ -13,3 +13,8 @@ module.exports = nuxifyStorybook({
     // Add your addons here
   ],
 })
+
+const generatedIconsStory = storybook.stories.indexOf("@nuxtjs/svg-sprite/stories/*.stories.js")
+storybook.stories[generatedIconsStory] = "../node_modules/@nuxtjs/svg-sprite/stories/*.stories.js"
+
+module.exports = storybook
```

That fixed Storybook for me locally.

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
LGTM. I have a couple clarifying questions but in the meantime the Storybook fix is the only requested change.

File: documentation/guides/frontend/icons.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+This guide explains how to use SVG sprites in Openverse Frontend and how to add
+a new SVG file to the sprite.
+
+## Using SVG Sprites in Openverse Frontend
+
+You can use SVG sprites through the `VIcon` component. To use it, first, import
+it into your component:
+
+```vue
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply documentation/guides/frontend/icons.md:18]: HTML seems to be a decent stand-in for Vue as noted in https://github.com/WordPress/openverse/issues/1810#issuecomment-1513189519.

File: frontend/nuxt.config.ts [hunk 4]
```diff
@@ -154,6 +154,7 @@ const config: NuxtConfig = {
     "@nuxtjs/composition-api/module",
     "@nuxtjs/style-resources",
     "@nuxtjs/svg",
+    "@nuxtjs/svg-sprite",
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/nuxt.config.ts:157]: I haven't tried replacing the non-icon files, but this is a great idea! I'll try.

[the remark this answers is not in the document]
Comment (obulat) [reply frontend/nuxt.config.ts:158]: I tried this locally, and removing the other SVG module would require adding a new component and handling sizes differently. I could open an issue to keep track of it.

File: frontend/test/unit/setup.js (hunk 1 above)
Comment (obulat) [reply frontend/test/unit/setup.js:28]: We can remove the transformer after removing the other SVG module. I'd prefer to do it in another PR.

**Review by sarayourfriend — APPROVED** [review]
LGTM once the playwright tests pass :slightly_smiling_face:

**Comment by obulat** [comment]
> Is there an issue to do that?

I'll open the issue once the Playwright tests pass :)

**Review by dhruvkb — APPROVED** [review]
LGTM! The added documentation is very helpful too.

**Pull request merged**